topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.md

```diff
@@ -0,0 +1,675 @@
+>    https://zenodo.org/record/7855968/files/atac_pbmc_1k_nextgem_S1_R2_001_chr21.fastq.gz
+>    https://zenodo.org/record/7855968/files/atac_pbmc_1k_nextgem_S1_R3_001_chr21.fastq.gz
+>    ```
+>
+>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
+>
+{: .hands_on}
+>
+> <question-title></question-title>
+> Inspect all the uploaded FASTQ files by clicking on the {% icon galaxy-eye %} symbol.
+> 1. Which FASTQ file contains the barcode sequence?
+> 1. What is the length of the barcode?
+>
+> > <solution-title></solution-title>
+> > 1. In this case, R2 contains the cell barcodes
+> > 1. 16bp
+> >
+> {: .solution}
+{: .question}
+
+
+## Attach cell barcodes to FASTQ headers
+
+First, we will attach the barcodes from the barcodes FASTQ file to the ids of forward and reverse reads that carry the association between reads and barcodes throughout the analysis. We will use this information to demultiplex the cells at the end while creating the count matrix. We will be left with only two FASTQ files of forward and reverse reads after this step.
+
+> <comment-title></comment-title>
+>
+> {% tool [Sinto barcode](toolshed.g2.bx.psu.edu/repos/iuc/sinto/sinto_barcode/0.9.0) %}  consumes the barcode sequence from the `atac_pbmc_1k_nextgem_S1_R2_001_chr21.fastq.gz` file and uses `:` delimiter to prepend it to both the forward and reverse read ids. The tool expects the same number of reads in all three FASTQ files and should be present in the same order.
+{: .comment}
+
+> <hands-on-title></hands-on-title>
+>
+> {% tool [Sinto barcode](toolshed.g2.bx.psu.edu/repos/iuc/sinto/sinto_barcode/0.9.0) %}  with the following parameters:
+>    - *"FASTQ file containing cell barcode sequences"*: `atac_pbmc_1k_nextgem_S1_R2_001_chr21.fastq.gz`
+>    - *"Single or Paired-end data"*: `Paired`
+>    - *"Forward reads FASTQ file"*: `atac_pbmc_1k_nextgem_S1_R1_001_chr21.fastq.gz`
+>    - *"Reverse reads FASTQ file"*: `atac_pbmc_1k_nextgem_S1_R3_001_chr21.fastq.gz`
+>    - *"Number of bases to extract from barcode-containing FASTQ"*: `16`
+>
+>    > <comment-title>Sinto barcode output</comment-title>
+>    >
+>    > After a successful run, the barcode sequences from the barcode FASTQ file are prepended to the read names. We can inspect the output FASTQ files by clicking on the {% icon galaxy-eye %} symbol of the *barcoded read 1* file. 
+>    {: .comment}
+>
+{: .hands_on}
+
+
+Now let's proceed with the next steps QC, mapping and peak calling as we do with the bulk ATAC-Seq data. For a thorough description of bulk ATAC-Seq analysis, please follow the bulk [ATAC-Seq data analysis tutorial]({% link topics/epigenetics/tutorials/atac-seq/tutorial.md %}). Here we use a slightly different workflow. Here we map reads using **BWA-MEM** instead of **Bowtie2**, then create ATAC fragments file that also stores the cell barcode information and finally use **MACS2** for peak calling. Note that the workflow from bulk ATAC-seq tutorial should also result in similar set of peaks.
```

Comment:
Here perhaps you could use BWA-MEM2 instead of BWA-MEM, since it generate similar results but it is faster.

Reply:
I ran BWA-MEM and BWA-MEM2 several times. BWA-MEM2 has a shorter CPU time but much longer (almost double) wall clock time. So BWA-MEM2 is taking longer to finish. I don't know why that is so. I will stick with the BWA-MEM for this version. We can update it to BWA-MEM2 in the later versions of the tutorial. Thanks for the recommendation!